test/functional/hosts_controller_test.rb

```diff
@@ -578,6 +578,10 @@ def test_unset_manage
     Setting[:restrict_registered_smart_proxies] = true
     Setting[:require_ssl_smart_proxies] = true
     SETTINGS[:require_ssl] = true
+    # Adding https and certificates or else the request will be redirected
+    @request.env['HTTPS'] = 'on'
```

Comment:
Since ```SETTINGS[:require_ssl] = true```, I have to simulate an https call, or else I'll be redirected to https (an the response will be 301, and not 403 as expected)